docs/markdown/snippets/jdk-system-dependency.md

```diff
@@ -0,0 +1,16 @@
+## JDK System Dependency
+
+When building projects such as those interacting with the JNI, you need access
+to a few header files located in a Java installation. This system dependency
+will add the correct include paths to your target. It assumes that either
+`JAVA_HOME` will be set to a valid Java installation, or the default `javac` on
+your system is a located in the `bin` directory of a Java installation. Note:
+symlinks are resolved.
+
+```meson
+jdk = dependency('jdk', version : '>=1.8', required : true)
```

Comment:
In case this needs any further changes, take out `required: true`. It's not needed as it is the default. However if this would be the only thing that would need changing then it's fine as is.

Reply:
I will push it up real quick.